### PR TITLE
feat: add support for selecting extensions

### LINKED ISSUE #388: Update design for adding a text file
screen shot 2017-07-11 at 12 49 42 pm
screen shot 2017-07-11 at 12 49 50 pm
screen shot 2017-07-11 at 12 50 00 pm
screen shot 2017-07-11 at 12 50 16 pm
screen shot 2017-07-11 at 12 50 26 pm

Get the details on [Zeplin](https://zpl.io/Z2cihf1).

### PR DESCRIPTION
**Pull request by dhruvdutt**
I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

Fixes #388

Screenshot 2019-03-28 at 2 21 27 AM

### CODE REVIEW

**Comment by catarak** [comment]
thanks for starting on this! i was looking at the [fileUtils.js](https://github.com/processing/p5.js-web-editor/blob/master/server/utils/fileUtils.js) file, and i think

File: client/modules/IDE/components/NewFileModal.jsx [hunk 1]
```diff
@@ -78,9 +80,16 @@ function validate(formProps) {
   return errors;
 }
 
+function mapDispatchToProps(dispatch) {
+  return {
+    setFieldValue: (fieldName, fieldValue) => {
+      dispatch(change('new-file', fieldName, fieldValue));
```
Comment (dhruvdutt) [inline client/modules/IDE/components/NewFileModal.jsx]: Is this the correct approach?

File: server/utils/fileUtils.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ export const fileExtensionsArray = ['gif', 'jpg', 'jpeg', 'png', 'bmp', 'wav', '
   'mpeg', 'mpe', 'mpv', 'js', 'jsx', 'html', 'htm', 'css', 'json', 'csv', 'obj', 'svg',
   'otf', 'ttf', 'txt', 'mov', 'vert', 'frag'];
 
+export const textFileExtensionsArray = ['json', 'txt', 'csv', 'vert', 'frag', 'js', 'css', 'html', 'htm', 'jsx'];
```
Comment (dhruvdutt) [inline server/utils/fileUtils.js]: This is the same as `TEXT_FILE_REGEX`. We need an array to iterate over the extensions list. Also, this list is different from the Zeplin design. Is this the final list of extensions?

Comment (catarak) [reply server/utils/fileUtils.js]: Yes! The list of extensions in the Zeplin design are just placeholders.

File: client/modules/IDE/components/NewFileModal.jsx (hunk 1 above)
Comment (catarak) [reply client/modules/IDE/components/NewFileModal.jsx]: no, you should use `bindActionCreators` from `redux`

**Comment by catarak** [comment]
There are so many file extension types that I no longer think this interface makes sense! Thank you for your work on this.

**Pull request closed without merging**